src/OpenApi/test/Services/OpenApiDocumentServiceTestsBase.cs

```diff
@@ -70,9 +84,30 @@ public TestEndpointRouteBuilder(IApplicationBuilder applicationBuilder)
         public IServiceProvider ServiceProvider => ApplicationBuilder.ApplicationServices;
     }
 
-    private class TestServiceProvider : IServiceProvider
+    private class TestServiceProvider : IServiceProvider, IKeyedServiceProvider
     {
         public static TestServiceProvider Instance { get; } = new TestServiceProvider();
+        internal OpenApiDocumentService TestDocumentService { get; set; }
+
+        public object GetKeyedService(Type serviceType, object serviceKey)
+        {
+            if (serviceType == typeof(OpenApiDocumentService))
+            {
+                return TestDocumentService;
+            }
+
+            return null;
+        }
+
+        public object GetRequiredKeyedService(Type serviceType, object serviceKey)
+        {
+            if (serviceType == typeof(OpenApiDocumentService))
+            {
+                return TestDocumentService;
+            }
+
+            return null;
```

Comment:
This should throw since it's required?

Reply:
This changed in [`73fc68a` (#54935)](https://github.com/dotnet/aspnetcore/pull/54935/commits/73fc68a58d1a8128004208eb86b9c79cc2b0d704) since we initialize this with a backup service provider.